### PR TITLE
ci: a bunch more backports to get MacOS job working

### PR DESCRIPTION
**Pull request by apoelstra**
Backport of https://github.com/bitcoin/bitcoin/pull/20419 then https://github.com/bitcoin/bitcoin/pull/19179 (except 2nd commit "Run i686 centos ci config on cirrus" which requires some more supporting PRs) then https://github.com/bitcoin/bitcoin/pull/20545 then https://github.com/bitcoin/bitcoin/pull/21542 then 
https://github.com/bitcoin/bitcoin/pull/21663 then https://github.com/bitcoin/bitcoin/pull/22122

We are getting 403 errors in our MacOS job on Cirrus, presumably because bintray.com has shut down (see discussion on upstream PR 21663). Backport this PR and several supporting ones. These PRs will be merged as part of the 22.0 rebase. In the interest of keeping CI running let's backport them early.

### CODE REVIEW

**Comment by apoelstra** [comment]
We actually need both this and #1017 to get CI working.

File: ci/test/00_setup_env_mac_host.sh [hunk 1]
```diff
@@ -10,7 +10,8 @@ export HOST=x86_64-apple-darwin18
 export PIP_PACKAGES="zmq"
 export GOAL="install"
 # ELEMENTS: add -fno-stack-check to work around clang bug on macos
-export BITCOIN_CONFIG="--with-gui --enable-reduce-exports --enable-werror --with-boost-process CXXFLAGS=-fno-stack-check"
+# ELEMENTS: remove --enable-werror because it triggers on Boost Thread includes (FIXME remove this after 22.0 rebase when boost-thread is removed)
```
Comment (stevenroose) [inline ci/test/00_setup_env_mac_host.sh:13]: Remembering this will be a non-trivial task :sweat_smile:

**Comment by stevenroose** [comment]
LGTM, I'm not fully understanding the OSX changes specifically, but the CI changes seem fine and I didn't see any other potential side-effects.

**Comment by stevenroose** [comment]
utACK 1e63e577db1a33226ba4412915fdfded28536e80

**Review by 0racl3z — COMMENTED** [review]
Thanks!

File: ci/test/00_setup_env_mac_host.sh (hunk 1 above)
Comment (apoelstra) [reply ci/test/00_setup_env_mac_host.sh:13]: In the 22 rebase I just blanked out this PR, since the PRs it backports preceded it.

**Pull request merged**